### DISCUSSION TITLE
Is there a way to save the generated Java source code?

### DISCUSSION THREAD

**Discussion by StSchnell** [mozilla/rhino#1441] [Q&A] [answered]
When I compile a JavaScript source code I get a class file.
With the command `java org.mozilla.javascript.tools.jsc.Main file.js` I get `file.class`
Is it also possible that the compiler save the Java source code? I mean that not only `file.class` but also `file.java` is available.
I am aware that I can also restore the Java source code from the class file using a decompiler. But my question is about getting the Java source code base directly from the Rhino compilation process.
Thanks for tips and answers.

**Comment by p-bakker** [comment, marked as answer]
Rhino doesn't generate any Java sourcecode, just bytecode, so no.

Only option is to put the generated .class file through a decompiler
